### PR TITLE
[ BTS & STB ] Fixing transformations

### PR DESCRIPTION
**Pull request by jane-intel**
Description: TF BTS/STB normalization transformation was inconsistent from shape and data type perspective

JIRA: 34153

**Code:**
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

**Validation:**
* [x]  Unit tests
* [N/A]  Framework operation tests
* [ ]  Transformation tests  -- **coverage was not lowered**
* [N/A]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py
* [N/A]  Model Optimizer IR Reader check

**Documentation:**
* [N/A]  Supported frameworks operations list
* [N/A]  Supported **public** models list
* [N/A]  New operations specification
* [N/A]  Guide on how to convert the **public** model
* [N/A]  User guide update

### CODE REVIEW

File: model-optimizer/extensions/front/tf/space_to_batch.py [hunk 1]
```diff
@@ -79,8 +81,7 @@ def find_and_replace_pattern(self, graph: Graph):
 
             in0_rank_1d = create_op_node_with_second_input(graph, Unsqueeze, int64_array([0]),
                                                            {'name': node.name + '/1d_rank_of_0'}, in0_rank)
-            in1_rank_1d = create_op_node_with_second_input(graph, Unsqueeze, int64_array([0]),
-                                                           {'name': node.name + '/1d_rank_of_1'}, in1_rank)
+            in1_rank_1d = in1_rank
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/tf/space_to_batch.py:84]: The `in1_rank` variable is not needed at all. The input of this node is connected in line 87 (using the name 'in1_rank') and the output of this node is connected in line 89 (using the name `in_rank_1d`). 

Moreover, this is not a rank node. It actually gets a shape from the input port 1 of the BTS/SPB node.

File: model-optimizer/extensions/front/tf/space_to_batch.py [hunk 2]
```diff
@@ -96,3 +97,5 @@ def find_and_replace_pattern(self, graph: Graph):
                 const_begin.out_port(0).connect(input_to_node.in_port(1))
                 diff.out_port(0).connect(input_to_node.in_port(2))
                 input_to_node.out_port(0).connect(node.in_port(idx + 1))
+                convert = Cast(graph, {'dst_type': np.int64}).create_node()
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/tf/space_to_batch.py:100]: Please, set the 'name' attribute for Node. Let's avoid random node names wherever possible.

File: model-optimizer/extensions/front/tf/space_to_batch.py [hunk 3]
```diff
@@ -60,9 +63,8 @@ def find_and_replace_pattern(self, graph: Graph):
 
             diff_size = Sub(graph, {'name': node.name + '/sub_0'}).create_node()
             diff = Sub(graph, {'name': node.name + '/sub_1'}).create_node()
-
             const_begin = Const(graph, {'value': int64_array([1])}).create_node()
-            const_pad_val = Const(graph, {'value': int64_array([1])}).create_node()
+            const_pad_val = Const(graph, {'value': int64_array(1)}).create_node()
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/tf/space_to_batch.py:67]: Implement a unit-test for this change.

**Comment by lazarevevgeny** [comment]
Please, update the PR description according to the template.

File: model-optimizer/extensions/front/tf/space_to_batch.py (hunk 3 above)
Comment (jane-intel) [reply model-optimizer/extensions/front/tf/space_to_batch.py:67]: I was surprised to see no transformation test for this replacement. So, I added an assert in the `Pad` layer to fail if it faces non-scalar 3rd input -- it is totally compliant with specification so the assertion is valid. And it fail model TF models conversion with BTS or STB if we wont perform this change. However, I filed a ticket to cover this transformation with transformation test: 34514. Is that OK?

Comment (lazarevevgeny) [reply model-optimizer/extensions/front/tf/space_to_batch.py:67]: Yes, looks good.

**Pull request merged**